scripts/lib/ts-generator.js

```diff
@@ -300,8 +320,10 @@ TSGenerator.prototype.generateTypingsFromShape = function generateTypingsFromSha
         code += tabs(tabCount) + 'export type ' + shapeKey + ' = Date;\n';
     } else if (type === 'boolean') {
         code += tabs(tabCount) + 'export type ' + shapeKey + ' = boolean;\n';
-    } else if (type === 'blob' || type === 'binary') {
-        code += tabs(tabCount) + 'export type ' + shapeKey + ' = Buffer|Uint8Array|Blob|string;\n';
+    } else if (type === 'blob' || type === 'binary') {     
+        code += tabs(tabCount) + 'export type ' + shapeKey + ' = Buffer|Uint8Array|Blob|string'+
+        (this.extraDependency[importStream] ? '|Readable' : '')
```

Comment:
This would add `Readable` to non-streaming blobs in services that support any streaming blobs. That appears to be happening with Lambda (where `BlobStream` should be a streaming shape but `Blob` should not).